### PR TITLE
Fix issues with searching for files on ESGF

### LINKED ISSUE #1861: File search on ESGF fails if dataset_id_template_ is incorrect
Example code
```python
from esmvalcore.esgf import find_files
find_files(project='obs4MIPs', short_name='*', dataset='*', version='*')
```

Last bit of the resulting stack trace:
```
File ~/src/esmvalgroup/esmvalcore/esmvalcore/esgf/_download.py:283, in ESGFFile._get_facets(results)
    279 facets = {
    280     'project': project,
    281 }
    282 for idx, key in enumerate(keys):
--> 283     facets[key] = values[idx]
    284 # The dataset_id does not contain the short_name for all projects,
    285 # so get it from the filename if needed:
    286 if 'short_name' not in facets:

IndexError: list index out of range
```

### LINKED ISSUE #1862: Searching ESGF with `short_name='*'` does not work
This returns no files.

Example code
```python
from esmvalcore.esgf import find_files
find_files(project='obs4MIPs', short_name='*', dataset='*', version='*')
```

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Closes #1861
Closes #1862

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/esgf/_download.py [hunk 1]
```diff
@@ -278,8 +278,12 @@ def _get_facets(results):
...
             'project': project,
         }
-        for idx, key in enumerate(keys):
-            facets[key] = values[idx]
+        if len(keys) == len(values):
+            for idx, key in enumerate(keys):
+                facets[key] = values[idx]
+        else:
+            logger.debug("Wrong dataset_id_template_ %s for dataset %s",
+                         template, dataset_id)
```
Comment (bouweandela) [inline esmvalcore/esgf/_download.py:286]: Maybe I should do something smarter here..

**Review by valeriupredoi — APPROVED** [review]
this looks good to me @bouweandela - unless you want to do something smarter in there where you say that?

**Comment by bouweandela** [comment]
Thanks @valeriupredoi! Let's get this merged and I'll make a new pull request to do something smarter ;-)

**Comment by schlunma** [comment]
Thanks :rocket:

File: esmvalcore/esgf/_download.py (hunk 1 above)
Comment (bouweandela) [reply esmvalcore/esgf/_download.py:286]: Continued in #1920

**Pull request merged**